src/braket/experimental/autoqasm/program/program.py

```diff
@@ -291,6 +344,33 @@ def validate_gate_targets(self, qubits: List[Any], angles: List[Any]) -> None:
                         "passed as arguments."
                     )
 
+    def _validate_verbatim_target_qubits(self, qubits: List[Any]) -> None:
+        # Only physical target qubits are allowed in a verbatim block:
+        for qubit in qubits:
+            if not isinstance(qubit, str):
+                qubit_name = qubit.name if isinstance(qubit, oqpy.Qubit) else str(qubit)
+                raise errors.InvalidTargetQubit(
+                    f'Qubit "{qubit_name}" is not a physical qubit. Only physical qubits such '
+                    'as "$0" can be targeted inside a verbatim block.'
+                )
+        qubits = physical_qubit_to_braket_qubit(qubits)
+
+        # Validate physical qubit connectivity on the target device:
+        device = self.get_target_device()
+        if device and not device.properties.paradigm.connectivity.fullyConnected:
+            # connectivityGraph uses integer qubit indices, but represented as strings.
+            connectivity_graph = device.properties.paradigm.connectivity.connectivityGraph
+            start_qubit = f"{int(qubits[0])}"
+            valid_target_qubits = connectivity_graph[start_qubit]
+            for qubit in qubits[1:]:
+                target_qubit = f"{int(qubit)}"
+                if target_qubit not in valid_target_qubits:
```

Comment:
Your understanding is correct. I actually don't know the desired behavior, because (I think) we don't have any examples of what the connectivity graph looks like for native 3-qubit gates. Imagine a native CCNOT gate - I'm not sure what the connectivity graph would look like for this. Do you know whether the behavior is specified somewhere?